app/addons/components/react-components.react.jsx

```diff
@@ -1160,6 +1171,61 @@ const ConfirmButton = React.createClass({
   }
 });
 
+
+// A simple, unstyled typeahead widget that can be used for typeahead on any arbitrary strings.
+// This should replace the more specific <JumpToDatabaseWidget />
+var TypeaheadField = React.createClass({
+  propTypes: {
+    list: React.PropTypes.array.isRequired,
+    value: React.PropTypes.string.isRequired,
+    onChange: React.PropTypes.func.isRequired,
+    placeholder: React.PropTypes.string,
+    autoFocus: React.PropTypes.bool
+  },
+
+  getDefaultProps: function () {
+    return {
+      placeholder: 'Type ahead',
+      autoFocus: false
+    };
+  },
+
+  componentDidMount: function () {
+    this.updateTypeahead();
+  },
+
+  updateTypeahead: function () {
+    $(ReactDOM.findDOMNode(this.refs.field)).typeahead({
```

Comment:
we are removing the jquery typeahed and are using react-select instead.

it is very easy to integrate, here is an example: https://github.com/apache/couchdb-fauxton/blob/5be261085761c5f1d0510f2d28117775ea3e0da2/app/addons/documents/components/jumptodoc.react.jsx